hook-worker/src/worker.rs

```diff
@@ -379,26 +389,35 @@ async fn send_webhook(
         .body(body)
         .send()
         .await
-        .map_err(|e| WebhookError::RetryableRequestError {
+        .map_err(|e| WebhookRequestError::RetryableRequestError {
             error: e,
+            response: None,
             retry_after: None,
         })?;
 
     let retry_after = parse_retry_after_header(response.headers());
 
-    match response.error_for_status() {
-        Ok(response) => Ok(response),
+    match response.error_for_status_ref() {
+        Ok(_) => Ok(response),
         Err(err) => {
             if is_retryable_status(
                 err.status()
                     .expect("status code is set as error is generated from a response"),
             ) {
-                Err(WebhookError::RetryableRequestError {
-                    error: err,
-                    retry_after,
-                })
+                Err(WebhookError::Request(
+                    WebhookRequestError::RetryableRequestError {
+                        error: err,
+                        response: response.text().await.ok(),
```

Comment:
I think it'd be a good idea, yeah, since it's a risk that could take down the service and put us into a crash loop (I think?).

Maybe like: https://github.com/PostHog/hog-rs/commit/ccd9cabeac5802b35c12627b03b9faa5bae5fdab

Reply:
For some reason the solution is not working. Even when setting `n` to `1`, we end up reading more (`8018`) than `n` bytes out of a total `20 * 10240` bytes.

I _think_ that `take` may be denoting how many chunks to read, not necessarily bytes. But I am not sure at all.

Reply:
I _think_ the problem is we are calling this `take`: https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.take. So the `take` method **on the stream**, not on `Bytes`.

Reply:
I can't think of a way to address this except iterating over chunks until we have `n` bytes in the buffer.

Reply:
@bretthoerner I've replaced the implementation with this idea :point_up:. See the test case I've added that is now passing and wasn't with the original implementation I cherry-picked from your commit.